utils/route_diag/src/main.cpp

```diff
@@ -0,0 +1,273 @@
+// Tool to test routing from one node to another node.
+//
+// This tool is intended to allow extremely verbose router output on exactly
+// one path to diagnose router behavior, e.g. look ahead and A* characteristics.
+//
+// Usage follows typically VPR invocation, except that no circuit is actually
+// used.  Instead --source_rr_node arguments is required, and either
```

Comment:
--source_rr_node argument (not arguments)?
I think it's also useful to say here that the tool can either route from one source rr_node to one sink rr_node and produce diagnostics, or (if --profile_source true) can profile from the specified source rr_node to all input pins in the device.  That partially overlaps the command line options help below, but I think gives useful context to the reader of this file, who might not find that help text and infer what it means.